src/main/java/io/tweag/sparkle/HaskellIterator.java

```diff
@@ -0,0 +1,55 @@
+package io.tweag.sparkle;
+
+import io.tweag.sparkle.SparkleBase;
+
+/**
+ * The elements of this iterator are obtained by calling successively a native
+ * function.
+ *
+ * SparkleBase is extended in order to ensure that the code of the native
+ * methods is loaded.
+ * */
+public class HaskellIterator<T> extends SparkleBase
+                                implements java.util.Iterator<T> {
+
+    /// This is a pointer to a function in a StablePtr.
+    private final long fun;
```

Comment:
Since field should be marked as `transient` to avoid it being serialized, since it's a pointer.